source/Applications/Advanced/DownsamplePointCloud.m

```diff
@@ -0,0 +1,150 @@
+% Import ZDF point cloud and downsample it.
+
+% The "Zivid3D.zdf" file has to be in the same folder as the "ReadZDF" file.
+Filename = 'Zivid3D.zdf';
+
+% Reading a .ZDF point cloud.
+[X,Y,Z,R,G,B,Image,Contrast] = zdfread(Filename);
+
+% Creating a point cloud object.
+XYZ(:,:,1) = X;
+XYZ(:,:,2) = Y;
+XYZ(:,:,3) = Z;
+pc=pointCloud(XYZ,'color',double(Image)./255);
+
+% Downsampling the point cloud.
+[X_new,Y_new,Z_new,Image_new] = downsample(X,Y,Z,Image,Contrast,4);
+XYZ_new(:,:,1) = X_new;
+XYZ_new(:,:,2) = Y_new;
+XYZ_new(:,:,3) = Z_new;
+pc_new = pointCloud(XYZ_new,'color',double(Image_new)./255);
+
+% Displaying RGB and Depth images.
+figure('units','normalized','outerposition',[0 0 1 1])
+subplot(2,2,1)
+imagesc(Image);
+subplot(2,2,2)
+imagesc(Image_new);
+subplot(2,2,3)
+imagesc(Z);
+colormap jet
+colorbar
+subplot(2,2,4)
+imagesc(Z_new);
+colormap jet
+colorbar
+pbaspect([1920 1200 1])
+
+%Visualizing the point clouds.
+figure('units','normalized','outerposition',[0 0 1 1])
+pcshow(pc);
+view([0 -90]);
+set(gca, 'visible', 'off')
+figure('units','normalized','outerposition',[0 0 1 1])
+pcshow(pc_new);
+view([0 -90]);
+set(gca, 'visible', 'off')
+
+function [X, Y, Z, R, G, B, Image, Contrast] = zdfread(Filename)
+
+% [X, Y, Z, R, G, B, Image, Contrast] = zdfread(Filename)
+%
+% Function for reading .ZDF files
+%
+% INPUT:
+% Filename - File to read
+%
+% OUTPUT:
+% X - x data in a matrix
+% Y - y data in a matrix
+% Z - z data in a matrix
+% R - Red color band from color image (scaled from 0 to 1)
+% G - Green color band from color image (scaled from 0 to 1)
+% B - Blue color band from color image (scaled from 0 to 1)
+% Image - Color image (uint8 - 0 to 255)
+% Contrast - Contrast image / quality image (float)
+
+try
+    if strcmp(Filename(end-3:end),'.zdf')
+        
+        disp(['Reading ',sprintf('%s',Filename),' point cloud'])
+        
+        % Reading the color image from rgba_image
+        % and spliting in R, G, B normalized matrices.
+        RGBA = ncread(Filename,'data/rgba_image');
+        RGBA = permute(RGBA,[3 2 1]);
+
+        R = double(RGBA(:,:,1)) / 255;
+        G = double(RGBA(:,:,2)) / 255;
+        B = double(RGBA(:,:,3)) / 255;
+
+        Image = uint8(RGBA(:,:,1:3));
+
+        % Reading the pointcloud data and creating X, Y, Z matrices.
+        Pointcloud = ncread(Filename,'data/pointcloud');
+        Pointcloud = permute(Pointcloud,[3 2 1]);
+
+        X = Pointcloud(:,:,1);
+        Y = Pointcloud(:,:,2);
+        Z = Pointcloud(:,:,3);
+
+        % Reading the contrast / quality data.
+        Contrast = ncread(Filename,'data/contrast')';
+    else
+        disp('Input file is not in ZDF format')
+    end
+catch ME
+    throw(ME)
+end
+
+end
+
+function [X_new, Y_new, Z_new, Image_new] = downsample(X, Y, Z, Image, Contrast, dsf)
```

Comment:
hasn't this been extracted into downsample.m?

Reply:
Opposite. I wanted to minmize the number of samples so I integrated downsample.m into DownsamplePointCloud.m.

I forgot to delete downsample.m - which I did now.